bin/git/hooks/pre-commit-golang

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+# This is a git commit-hook which will run the following tools against
+# any .go files in your repository:
+# - gofmt
+# - goimports
+# - golint
+# - go vet
+#
+# This script is meant to be added as a "pre-commit"-hook. See this
+#  page for further information:
+#   https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks
+#
+#
+
+gofiles=$(git diff --cached --name-only --diff-filter=ACM | grep '\.go$')
+[ -z "$gofiles" ] && exit 0
+
+gofmt -w ./..
+
+LINTERRORS=false
+VETERRORS=false
+
+for FILE in ${gofiles}
+do
+    goimports -w "${FILE}"
+    golint "-set_exit_status" "${FILE}"
+    if [[ $? == 1 ]]; then
+        LINTERRORS=true
+    fi
+
+    # vet everything except test files
+    if ! [[ "${FILE}" == *"_test.go"* ]]; then
+        go vet "${FILE}"
+        if [[ $? != 0 ]]; then
+            VETERRORS=true
+        fi
+    fi
```

Comment:
around here, we should add in `git add ${FILE}` I believe to restage these after the changes